tests/core/data_layer/test_data_rpc.py

```diff
@@ -421,3 +422,91 @@ async def test_get_root_history(one_wallet_node_and_rpc: nodes) -> None:
+
+        key1 = b"a"
+        value1 = b"\x01\x02"
+        changelist: List[Dict[str, str]] = [{"action": "insert", "key": key1.hex(), "value": value1.hex()}]
+        key2 = b"b"
+        value2 = b"\x03\x02"
+        changelist.append({"action": "insert", "key": key2.hex(), "value": value2.hex()})
+        key3 = b"c"
+        value3 = b"\x04\x05"
+        changelist.append({"action": "insert", "key": key3.hex(), "value": value3.hex()})
+        res = await data_rpc_api.batch_update({"id": store_id1.hex(), "changelist": changelist})
+        update_tx_rec0 = res["tx_id"]
+        await asyncio.sleep(1)
+        for i in range(0, num_blocks):
+            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+            await asyncio.sleep(0.2)
+        await time_out_assert(15, is_transaction_confirmed, True, "this is unused", wallet_rpc_api, update_tx_rec0)
+        history = await data_rpc_api.get_root_history({"id": store_id1.hex()})
+        diff_res = await data_rpc_api.get_kv_diff(
+            {
+                "id": store_id1.hex(),
+                "hash_1": bytes32([0] * 32).hex(),
+                "hash_2": history["root_history"][1]["root_hash"].hex(),
+            }
+        )
+        assert len(diff_res["diff"]) == 3
+        diff1 = {"type": "INSERT", "key": key1.hex(), "value": value1.hex()}
+        diff2 = {"type": "INSERT", "key": key2.hex(), "value": value2.hex()}
+        diff3 = {"type": "INSERT", "key": key3.hex(), "value": value3.hex()}
+        assert diff1 in diff_res["diff"]
+        assert diff2 in diff_res["diff"]
+        assert diff3 in diff_res["diff"]
+        key4 = b"d"
+        value4 = b"\x06\x03"
+        changelist = [{"action": "insert", "key": key4.hex(), "value": value4.hex()}]
+        key5 = b"e"
+        value5 = b"\x07\x01"
+        changelist.append({"action": "insert", "key": key5.hex(), "value": value5.hex()})
+        res = await data_rpc_api.batch_update({"id": store_id1.hex(), "changelist": changelist})
+        update_tx_rec1 = res["tx_id"]
+        await asyncio.sleep(1)
+        for i in range(0, num_blocks):
+            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+            await asyncio.sleep(0.2)
+        await time_out_assert(15, is_transaction_confirmed, True, "this is unused", wallet_rpc_api, update_tx_rec1)
+        history = await data_rpc_api.get_root_history({"id": store_id1.hex()})
+        diff_res = await data_rpc_api.get_kv_diff(
+            {
+                "id": store_id1.hex(),
+                "hash_1": history["root_history"][1]["root_hash"].hex(),
+                "hash_2": history["root_history"][2]["root_hash"].hex(),
+            }
+        )
+        assert len(diff_res["diff"]) == 2
+        diff4 = {"type": "INSERT", "key": key4.hex(), "value": value4.hex()}
+        diff5 = {"type": "INSERT", "key": key5.hex(), "value": value5.hex()}
+        assert diff4 in diff_res["diff"]
```

Comment:
probably not super important, but I would add a {type: "DELETE"} test here as well.